app/src/test/java/org/simple/clinic/user/UserSessionTest.kt

```diff
@@ -127,53 +132,110 @@ class UserSessionTest {
     Analytics.addReporter(reporter)
   }
 
+  private fun <T> unauthorizedHttpError(): HttpException {
+    val error = Response.error<T>(401, ResponseBody.create(MediaType.parse("text"), unauthorizedErrorResponseJson))
+    return HttpException(error)
+  }
+
   @Test
-  fun `login should correctly map network response to result`() {
-    whenever(loginApi.login(any()))
-        .thenReturn(Single.just(LoginResponse("accessToken", loggedInUserPayload)))
-        .thenReturn(Single.error(NullPointerException()))
-        .thenReturn(Single.error(unauthorizedHttpError<LoginResponse>()))
-        .thenReturn(Single.error(SocketTimeoutException()))
+  @Parameters(method = "params for mapping network response for login")
+  fun `login should correctly map network response to result`(
+      response: Single<LoginResponse>,
+      expectedResultType: Class<LoginResult>
+  ) {
+    whenever(loginApi.login(any())).thenReturn(response)
+    whenever(syncScheduler.syncImmediately()).thenReturn(Completable.complete())
+
+    val result = userSession.loginWithOtp("000000").blockingGet()
 
-    val result1 = userSession.loginWithOtp("").blockingGet()
-    assertThat(result1).isInstanceOf(LoginResult.Success::class.java)
+    assertThat(result).isInstanceOf(expectedResultType)
+  }
 
-    val result2 = userSession.loginWithOtp("").blockingGet()
-    assertThat(result2).isInstanceOf(LoginResult.UnexpectedError::class.java)
+  @Suppress("Unused")
+  private fun `params for mapping network response for login`(): List<List<Any>> {
+    return listOf(
+        listOf(Single.just(LoginResponse("accessToken", loggedInUserPayload)), LoginResult.Success::class.java),
+        listOf(Single.error<LoginResponse>(NullPointerException()), LoginResult.UnexpectedError::class.java),
+        listOf(Single.error<LoginResponse>(unauthorizedHttpError<LoginResponse>()), LoginResult.ServerError::class.java),
+        listOf(Single.error<LoginResponse>(IOException()), LoginResult.NetworkError::class.java)
+    )
+  }
 
-    val result3 = userSession.loginWithOtp("").blockingGet()
-    assertThat(result3).isInstanceOf(LoginResult.ServerError::class.java)
+  @Test
+  @Parameters(method = "parameters for triggering sync on login")
+  fun `sync should be triggered on successful login`(
```

Comment:
Should this be named as "sync should **only** be triggered on successful login"?